combine/combine.go

```diff
@@ -0,0 +1,226 @@
+
+		if err := keystore.StoreKeys([]tblsv2.PrivateKey{secret}, outPath); err != nil {
+			return errors.Wrap(err, "cannot store keystore", z.Int("validator_number", idx))
+		}
+	}
+
+	return nil
+}
+
+func secretsToShares(lock cluster.Lock, secrets []tblsv2.PrivateKey) (map[int]tblsv2.PrivateKey, error) {
+	n := len(lock.Operators)
+
+	resp := make(map[int]tblsv2.PrivateKey)
+	for idx, secret := range secrets {
+		pubkey, err := tblsv2.SecretToPublicKey(secret)
+		if err != nil {
+			return nil, errors.Wrap(err, "pubkey from share")
+		}
+
+		var found bool
+		for _, val := range lock.Validators {
+			for i := 0; i < n; i++ {
+				pubShare, err := val.PublicShare(i)
+				if err != nil {
+					return nil, errors.Wrap(err, "pubshare from lock")
+				}
+
+				if !bytes.Equal(pubkey[:], pubShare[:]) {
+					continue
+				}
+
+				resp[idx+1] = secret
+				found = true
+
+				break
+			}
+
+			if found {
+				break
+			}
+		}
+
+		if !found {
+			return nil, errors.New("share not found in lock")
+		}
+	}
+
+	return resp, nil
+}
+
+// loadLockfile loads a lockfile from one of the charon directories contained in dir.
+// It checks that all the directories containing a validator_keys subdirectory contain the same cluster_lock.json file.
+// It returns the cluster.Lock read from the lock file, and a list of directories that possibly contains keys.
+func loadLockfile(dir string) (cluster.Lock, []string, error) {
+	root, err := os.ReadDir(dir)
+	if err != nil {
+		return cluster.Lock{}, nil, errors.Wrap(err, "can't read directory")
+	}
+
+	lfFound := false
+	lastLockfileHash := [32]byte{}
+	lfContent := []byte{}
+	possibleValKeysDir := []string{}
+	for _, sd := range root {
+		if !sd.IsDir() {
+			continue
+		}
+
+		// try opening the lock file
+		lfPath := filepath.Join(dir, sd.Name(), "cluster-lock.json")
+		b, err := os.Open(lfPath)
+		if err != nil {
+			continue
+		}
+
+		// does this directory contains a "validator_keys" directory? if yes continue and add it as a candidate
+		vcdPath := filepath.Join(dir, sd.Name(), "validator_keys")
+		_, err = os.ReadDir(vcdPath)
+		if err != nil {
+			continue
+		}
+
+		possibleValKeysDir = append(possibleValKeysDir, vcdPath)
+
+		lfc, err := io.ReadAll(b)
+		if err != nil {
+			continue
+		}
+
+		lfHash := sha256.Sum256(lfc)
+
+		if lastLockfileHash != [32]byte{} && lfHash != lastLockfileHash {
+			return cluster.Lock{}, nil, errors.New("found different lockfile in node directory", z.Str("name", sd.Name()))
+		}
+
+		lastLockfileHash = lfHash
+		lfContent = lfc
+		lfFound = true
+	}
+
+	if !lfFound {
+		return cluster.Lock{}, nil, errors.New("lock file not found")
+	}
+
+	var lock cluster.Lock
+	if err := json.Unmarshal(lfContent, &lock); err != nil {
+		return cluster.Lock{}, nil, errors.Wrap(err, "unmarshal lock file")
+	}
+
+	if err := lock.VerifyHashes(); err != nil {
+		return cluster.Lock{}, nil, errors.Wrap(err, "cluster lock hash verification failed")
+	}
+
+	if err := lock.VerifySignatures(); err != nil {
```

Comment:
Should we allow `--no-verify` flags on this command @corverroos ? Imagined use case would be trying to recombine a really old cluster in a modern (>v1.0) charon version, and not being able to get it to verifySignatures correctly? (Or we can just pinky promise to always support old signatures/verification for anything that could possibly be in prod?)

Happy to skip this if not a concern

Reply:
I'm merging this PR, let's address this behavior in a separate one if needs a change.